lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -257,10 +256,13 @@ def main():
     required_together = [['s3_key', 's3_bucket'],
                          ['vpc_subnet_ids', 'vpc_security_group_ids']]
 
+    required_if = [['state', 'present', ['runtime', 'handler', 'role']]]
```

Comment:
I don't like the idea of it being easy to make incomplete definitions which seem to work when the lambda is there but then fail when it disappears.  I think ansible should generally have a full declarative view of the system and update it to match.  OTOH maybe this is a useful feature for operational procedures, so maybe there should be a separate "update" mode??